src/plugins/index_pattern_field_editor/public/components/field_editor/field_editor.tsx

```diff
@@ -194,6 +200,9 @@ const FieldEditorComponent = ({
     }
   }, [onChange, isFormValid, isSubmitted, submit]);
 
+  const [nameChanged, setNameChanged] = useState(false);
```

Comment:
Instead of using the `onChange` handlers can you try this (not tested)

```js
const [{ name: updatedName, type: updatedType }] = useFormData({ form });
  const nameHasChanged = Boolean(field?.name) && field?.name !== updatedName;
  const typeHasChanged = Boolean(field?.type) && field?.type !== updatedType;
```

And update L245 accordingly?